server/lib/renderers/xlsxReceipt.js

```diff
@@ -0,0 +1,207 @@
+
+
+  ws.cell(7, 5, 7, 7, true).string(t('FORM.LABELS.INVOICE').concat(': ', data.reference));
+  ws.cell(8, 5, 8, 7, true).string(t('TABLE.COLUMNS.SERVICE').concat(': ', data.serviceName));
+  ws.cell(9, 5, 9, 7, true).string(t('FORM.LABELS.DATE').concat(': ', data.dateFormat));
+  ws.cell(10, 5, 10, 7, true).string(t('FORM.LABELS.CREATED_BY').concat(': ', data.display_name));
+
+  ws.cell(7, 1, 10, 7).style(style2);
+
+
+  ws.cell(7, 1, 7, 7).style(styleBorderTop);
+
+  ws.cell(10, 1, 10, 7).style(styleBorderBottom);
+  ws.cell(7, 1, 10, 1).style(styleBorderLeft);
+  ws.cell(7, 7, 10, 7).style(styleBorderRight);
+
+  // Description
+
+  ws.cell(12, 1).string(t('FORM.LABELS.DESCRIPTION')).style(styleUndeLine);
+  ws.cell(13, 1, 13, 7, true).string(data.description).style(style2);
+  ws.cell(13, 1, 13, 7, true).style(styleMultiLine);
+  ws.row(13).setHeight(40); // description might be pretty long
+
+  // detail
+  ws.cell(15, 1).string(t('FORM.LABELS.INVOICES_DETAILS')).style(styleUndeLine);
+
+  // titles
+  let line = 16;
+  ws.cell(line, 1).string(t('TABLE.COLUMNS.CODE'));
+  ws.cell(line, 2, line, 4, true).string(t('TABLE.COLUMNS.DESCRIPTION'));
+  ws.cell(line, 5).string(t('TABLE.COLUMNS.UNIT_PRICE'));
+  ws.cell(line, 6).string(t('TABLE.COLUMNS.QUANTITY'));
+  ws.cell(line, 7).string(t('TABLE.COLUMNS.TOTAL'));
+  ws.cell(line, 1, line, 7).style(style2);
+  ws.cell(line, 1, line, 7).style(styleAllBorders);
+
+  line = 17;
+  // invoice items
+  data.items.forEach(row => {
+    ws.cell(line, 1).string(row.code);
+    ws.cell(line, 2, line, 4, true).string(row.text);
+    ws.cell(line, 5).number(row.transaction_price);
+    ws.cell(line, 6).number(row.quantity);
+
+    const total = (row.transaction_price * row.quantity);
+    ws.cell(line, 7).string(''.concat(total, '', currencySymbol));
+
+    ws.cell(line, 1, line, 7).style(style2);
+    ws.cell(line, 1, line, 7).style(styleAllBorders);
+    line++;
+  });
+
+  // subsidy
+
+  if (data.subsidy.length) {
+    const subsidyLable = t('FORM.LABELS.SUBSIDIES').concat('(', data.subsidy.length, ')');
+    ws.cell(line, 3, line, 4, true).string(subsidyLable).style(style2);
+    ws.cell(line, 5, line, 7, true).number(math.sum(data.subsidy, 'value')).style(style1);
+    ws.cell(line, 3, line, 7, true).style(styleAllBorders);
+  }
+  // Total
+  line++;
+  ws.cell(line, 3, line, 4, true).string(t('FORM.LABELS.TOTAL')).style(style1);
+  ws.cell(line, 5, line, 7, true).number(data.cost).style(style1);
+  ws.cell(line, 3, line, 7, true).style(styleAllBorders);
+  line++;
+
+  ws.cell(line, 3, line, 7, true).string(numberToText.convert(data.cost, data.lang, currencyName)).style(style1);
+  ws.cell(line, 3, line, 7, true).style(styleAllBorders);
+  line++;
+
+  const deferred = q.defer();
+
+  wb.writeToBuffer().then(xlsxBuffer => {
+    deferred.resolve(xlsxBuffer);
+  });
+  return deferred.promise;
```

Comment:
There is no need to do this - in fact, you will lose errors from this if `wb.writeToBuffer()` errors since you have no catch statement.

Just return the `wb.writeToBuffer()`.  Like this:
```js
return wb.writeToBuffer();
```